File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string str;
        for(int i=0;i<s.length();i++){
            if(s[i]>='A' and s[i]<='Z'){
                s[i] = s[i]-'A'+'a';
            }
        }
        for(int i=0;i<s.length();i++){
            if((s[i]>='a' and s[i]<='z') or (s[i]>='0' and s[i]<='9')){
                str.push_back(s[i]);
            }
        }
        cout<<str;
        int start = 0,end=str.length()-1;
        while(start<=end){
            if(str[start] != str[end]){
                // cout<<start<<" "<<end<<endl;
                // cout<<s[start]<<" "<<s[end];
                return false;
            }
            start++;end--;
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder sb = new StringBuilder();
        int n = s.length();

        for(int i=0; i<n; i++){
            char c = s.charAt(i);
            if('a' <= c && c <= 'z'){
                sb.append(c);
            }else if('A' <= c && c <= 'Z'){
                int x = (int)(c - 'A');
                char z = (char)('a'+ x);
                sb.append(z);
            }else if('0' <= c && c <= '9'){
                sb.append(c);
            }
        }
        System.out.print(sb);
        int i=0, j = sb.length()-1;
        while(i<j){
            if(sb.charAt(i) != sb.charAt(j)){
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
}